src/codeu/chat/client/commandline/Chat.java

```diff
@@ -40,6 +43,44 @@ public Chat(Context context) {
 		this.panels.push(createRootPanel(context));
 	}
 
+	// RECREATE 
+	//
+	// Recreates the server's last known state to display for the client 
+	// upon relaunching the client 
+	// Will not display anything if there was no information saved 
+	//
+	//
+	public void recreate(Context context) {
+		Hashtable<Uuid, String> allUsers = new Hashtable<Uuid, String>();
+
+		if (context.allUsers().iterator().hasNext()) {
+
+			Iterator<UserContext> users = context.allUsers().iterator(); 
+			UserContext user;
+			System.out.format("USERS: \n");
+			while (users.hasNext()){
+				user = users.next(); 
+				System.out.format("\t\"%s\" Added at %s || UUID: %s\n", 
```

Comment:
I ended up doing both just so that it's easily viewable to the user in the client window and it's also saved in a log file so it's easy to look through if we need to for debugging purposes.